src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -88,14 +88,23 @@ export class SharedCredentialsProvider implements CredentialsProvider {
     }
 
     public async getCredentials(): Promise<AWS.Credentials> {
-        const validationMessage = this.validate()
-        if (validationMessage) {
-            throw new Error(`Profile ${this.profileName} is not a valid Credential Profile: ${validationMessage}`)
-        }
+        const originalStsConfiguration = AWS.config.sts
+
+        try {
+            const validationMessage = this.validate()
+            if (validationMessage) {
+                throw new Error(`Profile ${this.profileName} is not a valid Credential Profile: ${validationMessage}`)
+            }
 
-        const provider = new AWS.CredentialProviderChain([this.makeCredentialsProvider()])
+            const provider = new AWS.CredentialProviderChain([this.makeCredentialsProvider()])
 
-        return provider.resolvePromise()
+            // Profiles with references involving non-aws partitions need help getting the right STS endpoint
+            this.applyProfileRegionToGlobalStsConfig()
```

Comment:
I feel like this was the crux of the complexity for the JetBrains implementation of partitions. Does this mean only one STS region can be configured at least one time? Why wouldn't you have one per partition.

Reply:
For the toolkit to have one "go-to" region per partition for STS, it means two things:
1. we'd have to define a toolkit lookup, and maintain it as endpoints evolve
2. we'd have to resolve the credentials' region to a partition, and then over to the toolkit lookup
I considered this, but it didn't seem necessary.